src/ServiceControl.Audit.Persistence.PostgreSQL/PostgreSQLPersistenceInstaller.cs

```diff
@@ -0,0 +1,238 @@
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_critical_time ON processed_messages (
+                critical_time
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_processing_time ON processed_messages (
+                processing_time
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_delivery_time ON processed_messages (
+                delivery_time
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_message_id ON processed_messages (
+                message_id
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_created_at ON processed_messages (
+                created_at
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_conversation ON processed_messages (
+                conversation_id,
+                time_sent
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_query ON processed_messages (
+                query
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_audit_counts ON processed_messages (
+                receiving_endpoint_name, processed_at
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        // Create saga_snapshots table
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE TABLE IF NOT EXISTS saga_snapshots (
+                id UUID PRIMARY KEY,
+                saga_id UUID,
+                saga_type TEXT,
+                changes JSONB,
+                updated_at TIMESTAMPTZ NOT NULL DEFAULT now()
+            );", connection))
```

Comment:
The design of this table is incorrect, needs to be updated to the same design as the know_endpoints where we use an insert only table for the hot path and then we use a background job to reconcile the results in the more permanent table.